### PR TITLE
Feature/split get service and workflows

### PR DESCRIPTION
**Pull request by agduncan94**
Before there was an endpoint userWorkflows that grabs all BioWorkflows and Services for a user. userWorkflows now only returns BioWorkflows, and Services can be retrieved by userServices endpoint.

### CODE REVIEW

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/ServiceIT.java [hunk 1]
```diff
@@ -185,6 +186,13 @@ public void testGitHubAppEndpoints() throws Exception {
         final long count = testingPostgres
                 .runSelectStatement("select count(*) from service where sourcecontrol = 'github.com' and organization = 'DockstoreTestUser2' and repository = 'test-service'", new ScalarHandler<>());
         Assert.assertEquals("there should be one matching service", 1, count);
+
+        // Test user endpoints
+        UsersApi usersApi = new UsersApi(webClient);
+        List<io.swagger.client.model.Workflow> services = usersApi.userServices(service.getUsers().get(0).getId());
+        List<io.swagger.client.model.Workflow> workflows = usersApi.userWorkflows(service.getUsers().get(0).getId());
+        assertTrue("There should be one service", services.size() == 1);
```
Comment (garyluu) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/ServiceIT.java]: assertEqual

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java [hunk 2]
```diff
@@ -498,7 +500,25 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
...
+
+    @GET
+    @Path("/{userId}/services")
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @ApiOperation(value = "List all services owned by the logged-in user.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Workflow.class, responseContainer = "List")
+    public List<Workflow> userServices(@ApiParam(hidden = true) @Auth User user,
+            @ApiParam(value = "User ID", required = true) @PathParam("userId") Long userId) {
+        checkUser(user, userId);
+        final User authUser = this.userDAO.findById(userId);
```
Comment (garyluu) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java]: I find it more intuitive if the `@Auth User user` is `@Auth User authUser` and this one is just `final User user` because the 2nd one can really be any user if the first one is an admin

File: dockstore-webservice/src/main/resources/openapi.yaml [hunk 3]
```diff
@@ -4346,6 +4346,32 @@ paths:
...
+            format: int64
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Workflow"
```
Comment (garyluu) [inline dockstore-webservice/src/main/resources/openapi.yaml]: Might run into problems from the UI side if there are properties that `Service` has that `Workflow` does not.  Same with BioWorkflow

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java [hunk 4]
```diff
@@ -491,14 +493,32 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
...
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java]: If using fetchedUser above, should probably use it here too

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java [hunk 5]
```diff
@@ -491,14 +493,32 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
...
     private List<Workflow> getWorkflows(User user) {
-        return user.getEntries().stream().filter(Workflow.class::isInstance).map(Workflow.class::cast).collect(Collectors.toList());
+        return user.getEntries().stream().filter(BioWorkflow.class::isInstance).map(BioWorkflow.class::cast).collect(Collectors.toList());
+    }
+
+    @GET
+    @Path("/{userId}/services")
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @ApiOperation(value = "List all services owned by the logged-in user.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Workflow.class, responseContainer = "List")
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java]: Let's also start to systematically set the operationId (nickname) to all endpoints due to https://github.com/anvilproject/client-apis/issues/8 kind of thing

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml [hunk 6]
```diff
@@ -1307,7 +1309,7 @@ paths:
           description: default response
   /api/ga4gh/v2/extended/containers/{organization}:
     get:
-      operationId: entriesOrgGet
+      operationId: entriesOrgGet_1
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/resources/openapi3/openapi.yaml]: Not really part of the PR, but hunt down and set a (unique) operation id for this and the one below to stop them from bouncing back and forth.

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java (hunk 5 above)
Comment (agduncan94) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java]: Would we name the operationId similarly to how we name the function?

Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java]: I have no strong preference for new endpoints (although it would be nice to use camelcase like existing functions since the operation id determines the method name in codegen).
For existing methods might as well lock it down to the existing, although since we have duplicates here https://github.com/anvilproject/client-apis/pull/9 it might be nice to avoid conflicts

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java [hunk 7]
```diff
@@ -491,14 +493,32 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
     public List<Workflow> userWorkflows(@ApiParam(hidden = true) @Auth User user,
             @ApiParam(value = "User ID", required = true) @PathParam("userId") Long userId) {
         checkUser(user, userId);
-        final User authUser = this.userDAO.findById(userId);
-        List<Workflow> workflows = getWorkflows(authUser);
+        final User fetchedUser = this.userDAO.findById(userId);
+        List<Workflow> workflows = getWorkflows(fetchedUser);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java]: I think you need to check for a null fetched user; could be an incorrect id.
